deepinv/physics/phase_retrieval.py

```diff
@@ -318,3 +317,227 @@ def get_structure(n_layers) -> str:
         :return: (str) the structure of the operator, e.g., "FDFD".
         """
         return "FD" * math.floor(n_layers) + "F" * (n_layers % 1 == 0.5)
+
+
+class PtychographyLinearOperator(LinearPhysics):
+    r"""
+    Forward linear operator for phase retrieval in ptychography. Modelling multiple applications of the shifted probe and Fourier transform on an input image.
+
+    This operator performs multiple 2D Fourier transforms on the probe function applied to the shifted input image according to specific offsets, and concatenates them.
+    The probe function is applied element by element to the input image.
+
+    .. math::
+
+        B = \left[ \begin{array}{c} B_1 \\ B_2 \\ \vdots \\ B_{n_{\text{img}}} \end{array} \right],
+        B_l = F P T_l, \quad l = 1, \dots, n_{\text{img}},
+
+    where :math:`F` is the 2D Fourier transform, :math:`P` is the probe function and :math:`T_l` is a shift.
+
+    :param tuple img_size: Shape of the input image (height, width).
+    :param probe: A 2D tensor representing the probe function.
+    :param str probe_type: Type of probe (e.g., "disk"), used if `probe` is not provided.
```

Comment:
```suggestion
    :param str probe_type: Type of probe (e.g., "disk"), used if ``probe`` is not provided.
```

restructured text is different than markdown, you need to use double accents to have code format.